File: LeetCode/LeetCode 908. 最小差值 I.py
class Solution:
    def smallestRangeI(self, A: List[int], K: int) -> int:
        maxn = max(A)
        minx = min(A)

        if maxn - minx <= 2 * K:
            return 0
        else:
            return maxn - minx - 2 * K

File: LeetCode/LeetCode 908. 最小差值 I.cpp
class Solution {
public:
    int smallestRangeI(vector<int>& A, int K) {
        int maxn = INT_MIN, minx = INT_MAX;
        for(auto &x : A) {
            maxn = max(maxn, x);
            minx = min(minx, x);
        }
        maxn -= minx;
        if(maxn <= 2 * K) maxn = 0;
        else maxn -= 2 * K;
        return maxn;
    }
};